openapi/src/main/java/io/helidon/openapi/OpenAPISupport.java

```diff
@@ -630,24 +656,35 @@ static SEOpenAPISupportBuilder builderSE() {
      * Helidon SE apps and once for use from the Helidon MP-provided OpenAPI
      * service. This lets us constrain what use cases are possible from each
      * (for example, no anno processing from SE).
+     *
+     * @param <T> concrete subclass of OpenAPISupport
+     * @param <B> concrete subclass of OpenAPISupport.Builder
      */
-    public abstract static class Builder implements io.helidon.common.Builder<OpenAPISupport> {
+    public abstract static class Builder<T extends OpenAPISupport<T, B>, B extends Builder<T, B>>
+            implements io.helidon.common.Builder<OpenAPISupport> {
 
         /**
          * Config key to select the openapi node from Helidon config.
          */
         public static final String CONFIG_KEY = "openapi";
 
+        private final Class<B> builderClass;
+
         private Optional<String> webContext = Optional.empty();
         private Optional<String> staticFilePath = Optional.empty();
         private CrossOriginConfig crossOriginConfig = null;
 
-        @Override
-        public OpenAPISupport build() {
-            validate();
-            return new OpenAPISupport(this);
+        protected Builder(Class<B> builderClass) {
+            this.builderClass = builderClass;
+        }
+
+        protected B me() {
+            return builderClass.cast(this);
         }
 
+        @Override
+        public abstract T build();
```

Comment:
This can be removed (already implied by `implements io.helidon.common.Builder<T>`)

Reply:
Fixed.